R/rep.R

```diff
@@ -5,10 +5,22 @@
 #'
 #' - `vec_rep_each()` repeats each element of a vector a set number of `times`.
 #'
+#' - `vec_unrep()` compresses a vector with repeated values. The repeated values
+#'   are returned as a `key` alongside the number of `times` each key is
+#'   repeated.
+#'
 #' @details
-#' `vec_rep()` and `vec_rep_each()` work along the size of `x`, rather than
-#' its length. For data frames, this means that rows are repeated rather
-#' than columns.
+#' These functions work along the size of `x`, rather than its length. For data
+#' frames, this means that rows are repeated rather than columns.
+#'
+#' Using `vec_unrep()` and `vec_rep_each()` together is very similar to using
+#' [base::rle()] and [base::inverse.rle()]. The following invariant shows
+#' the relationship between the two functions:
+#'
+#' ```
+#' compressed <- vec_unrep(x)
+#' x == vec_rep_each(compressed$key, compressed$times)
```

Comment:
The `==` is a bit confusing because of the vectorised R operation. Perhaps `identical()` or `===`?